### ISSUE TITLE
[FEA] Switch `orc_read_decode`, `parquet_read_decode` and `csv_read_input` to use DEVICE_BUFFER data source  

### ISSUE THREAD

**Issue by GregoryKimball** [rapidsai/cudf#12678]
**Is your feature request related to a problem? Please describe.**

The ORC reader benchmark `orc_read_decode` uses a [HOST_BUFFER](https://github.com/rapidsai/cudf/blob/b6dccb31c32eefbf29a7f731b4b5cd7d7f07fe0d/cpp/benchmarks/io/orc/orc_reader_input.cpp#L75) data source, but the Parquet reader benchmark `parquet_read_decode` uses a [FILEPATH](https://github.com/rapidsai/cudf/blob/b6dccb31c32eefbf29a7f731b4b5cd7d7f07fe0d/cpp/benchmarks/io/parquet/parquet_reader_input.cpp#L69) data source. The CSV `BM_csv_read_input` uses a FILEPATH data source.

Let's switch them all to DEVICE_BUFFER data sources.

Let's please also modify these benchmarks to accept an nvbench axis - so we can change the data source at runtime via the nvbench CLI.

**Describe the solution you'd like**
I'd like to be consistent and use a DEVICE_BUFFER data source for decode and data type benchmarking.

**Describe alternatives you've considered**
We considered choosing FILEPATH or HOST_BUFFER for all of these benchmarks, but DEVICE_BUFFER is best for focusing on decoding.

**Additional context**
I wrote a detailed issue full of profiles before realizing the problem. 🤦

**Comment by GregoryKimball** [comment]
@vuule

**Comment by vuule** [comment]
Sorry for the late comment.
The group of benchmarks in question are meant to be used to evaluate decode performance, hence the use of HOST_BUFFER. IMO the corresponding Parquet benchmarks should use a buffer source as well. I even think they should use a device buffer as the source, to further reduce impact of data copies.
I understand that these benchmarks have a gap (file IO + specific type), but I would like us to consider a different way to meet the requirements, and leave these benchmarks focused on the decode.
TL;DR I'd change both ORC and Parquet decode benchmarks to use DEVICE_BUFFER, but this causes practical issues for Greg's charts. Let's figure it out :)
